src/gateway/sharding/shard_manager.rs

```diff
@@ -390,14 +379,70 @@ pub struct ShardManagerOptions {
     #[cfg(feature = "voice")]
     pub voice_manager: Option<Arc<dyn VoiceGatewayManager>>,
     pub ws_url: Arc<str>,
+    pub compression: TransportCompression,
     pub shard_total: NonZeroU16,
+    pub max_concurrency: NonZeroU16,
+    pub wait_time_between_shard_start: Duration,
     #[cfg(feature = "cache")]
     pub cache: Arc<Cache>,
     pub http: Arc<Http>,
     pub intents: GatewayIntents,
     pub presence: Option<PresenceData>,
-    pub max_concurrency: NonZeroU16,
-    /// Number of seconds to wait between starting each shard/set of shards start
-    pub wait_time_between_shard_start: Duration,
-    pub compression: TransportCompression,
+}
+
+pub enum ShardManagerMessage {
+    PollShardQueue,
```

Comment:
Can this variant be documented, or even document all the variants? You know they need explanation as you did it in the PR description.

Reply:
Done. Renamed the `Err` variant to `Quit` as well.